### ISSUE TITLE
Docs: Documents state that `--ext` is no longer supported, but do not say what it is replaced by / how it works now.

### ISSUE THREAD

**Issue by Standard8** [eslint/eslint#17931]
### Docs page(s)

https://eslint.org/docs/latest/use/command-line-interface#--ext
https://eslint.org/docs/next/use/migrate-to-9.0.0
https://eslint.org/docs/latest/use/configure/migration-guide#cli-flag-changes

### What documentation issue do you want to solve?

When running the 9.0.0 alpha 0 version of ESLint, I get:

```
Invalid option '--ext' - perhaps you meant '-c'?
You're using eslint.config.js, some command line flags are no longer available. Please see https://eslint.org/docs/latest/use/command-line-interface for details.
```

Visiting the page states that `--ext` is `eslintrc Mode Only` but doesn't state how it has been replaced and what I have to do to make sure it still functions the same.

The main 9.0.0 migration page also doesn't list the change - though this [duplicate(?) page](https://eslint.org/docs/latest/use/configure/migration-guide#cli-flag-changes) lists it as being removed, but still doesn't specify the replacement.

### What do you think is the correct solution?

I think adding descriptions that indicate the files are now detected from the configuration, and that also points to https://eslint.org/docs/latest/use/configure/configuration-files-new

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by fasttime** [comment]
I think that https://eslint.org/docs/latest/use/configure/migration-guide is supposed to help migrating from the legacy config to the new config system, while https://eslint.org/docs/next/use/migrate-to-9.0.0 is specific for migrating from ESLint v8 to v9. Granted, some content may be overlapping or not where expected.

**Comment by fasttime** [comment]
The `--ext` CLI flag is peculiar to the old config system, so it seems correct to document it in [that paragraph](https://eslint.org/docs/latest/use/configure/migration-guide#cli-flag-changes).

I also think that it would be useful to clarify how that flag can be replaced in the new config, and how the file extensions are matched.

**Comment by nzakas** [comment]
> I think that https://eslint.org/docs/latest/use/configure/migration-guide is supposed to help migrating from the legacy config to the new config system, while https://eslint.org/docs/next/use/migrate-to-9.0.0 is specific for migrating from ESLint v8 to v9. Granted, some content may be overlapping or not where expected.

That's correct. We provided migration guides separately for configs and plugins because people could start using flat config in v8.x.

@Standard8 `--ext` doesn't work with flat config at all. You should specify the file patterns you want in your `eslint.config.js` file.
So if you did `--ext .jsx` before, you'd need to have a config that looks like this:

```js
export default [
  { files: ["**/*.jsx"] }
];
```

Then when you run something like `npx eslint .`, it will find the `.jsx` files.

I'll update the docs.

**Comment by slhck** [comment]
I'm sorry to revive this old issue, but I still cannot understand what to do based on simply upgrading eslint and looking at:

https://eslint.org/docs/latest/use/migrate-to-9.0.0

There is no mention of `--ext` here.

**Comment by nzakas** [comment]
@slhck Please see https://eslint.org/docs/latest/use/configure/migration-guide#--ext

If you are still having problems, please open a Discussion.
